### DISCUSSION TITLE
Convert Slider to Stream in osu! Editor with variable Spacing adjustment

### DISCUSSION THREAD

**Discussion by MaxKruse** [ppy/osu#22110] [Ideas] [answered]
I am aware this was mentionedin https://github.com/ppy/osu/discussions/21229 but for sake of simplicity/not bulk requesting, this post is made.

In short, lazer's functionality to convert streams to sliders seems rather blant and non-functional, as it only converts the slider's path with a fixed amount of 1/4 timing, and equal spacing gaps to fill the time the slider initially covered (e.g., a 1 second slider that covers 4 x 1/4 in the timeline will convert into 4 circles, equally paced).

I think additional parameters for converting sliders into streams are good to bring the editor up to modern needs.

One idea I had in mind was to place the settings below the DistanceSnap option on the right-hand side of the editor (as opposed to a right-click menu that would get crowded with options) while a slider is selected. There, the following could be useful:

1. Amount of circles
2. Distance Spacing (these 2 mimicking stable's settings for converting, essentially blocking each other like radioboxes)
3. Spacing Multiplier with different algorithms (e.g. linear, square, log etc.) To facilitate increasing-spacing streams (>1.0x multiplier), or streams that instead reduce in spacing over time (<1.0x multiplier)

If this is something that would be wanted, I'd also be willing to implement this.

**Comment by peppy** [comment, marked as answer]
Timing is based on your beat snap divisor. You can adjust top-right.

Also, a stream creation tool is coming soon, which will cover most of what you want. Please look forward to that!
